Source: dleverton
Upstream: no; one assumes they mangle CFLAGS on purpose
Reason: don't mess with CFLAGS just because the debug option is on

Index: grcompiler/configure.ac
===================================================================
--- grcompiler/configure.ac	(revision 815)
+++ grcompiler/configure.ac	(working copy)
@@ -14,8 +14,7 @@
 
 # Optional features
 AC_ARG_ENABLE(debug,
-    AC_HELP_STRING([--enable-debug], [build with debugging. (default=no)]),
-    [enable_debug=yes], [enable_debug=no])
+    AC_HELP_STRING([--enable-debug], [build with debugging. (default=no)]))
 AC_ARG_ENABLE(final,
     AC_HELP_STRING([--enable-final], [build with optimizations and no debugging. (default=no)]),
     [enable_final=yes], [enable_final=no])
@@ -51,13 +50,9 @@
 
 # Setup the build compiler flags and linker flags.
 # Enable debug
-if test "$enable_debug" != no; then
-    CFLAGS="$CFLAGS -O0"
-    CXXFLAGS="$CXXFLAGS -O0"
-    test "$ac_cv_prog_cc_g" = yes     && CFLAGS="$CFLAGS -g"
-    test "$ac_cv_prog_cxx_g" = yes    && CXXFLAGS="$CXXFLAGS -g"
-    CFLAGS="$CFLAGS -Wno-unknown-pragmas -Wparentheses -Werror "
-    CXXFLAGS="$CXXFLAGS -Wno-unknown-pragmas -Wparentheses -Werror "
+if test "$enable_debug" = yes; then
+    CFLAGS="$CFLAGS -D_DEBUG"
+    CXXFLAGS="$CXXFLAGS -D_DEBUG"
 else
     CFLAGS="$CFLAGS -DNDEBUG"
     CXXFLAGS="$CXXFLAGS -DNDEBUG"
